test/timeout_test.dart

```diff
@@ -2,24 +2,126 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:async';
+
+import 'package:grpc/grpc.dart';
+import 'package:http2/transport.dart';
+import 'package:test/src/backend/invoker.dart';
 import 'package:test/test.dart';
 
 import 'package:grpc/src/client.dart';
 
+import 'src/client_utils.dart';
+import 'src/server_utils.dart';
+import 'src/utils.dart';
+
 void main() {
-  test('Timeouts are converted correctly', () {
-    expect(ClientCall.toTimeoutString(new Duration(microseconds: -1)), '1n');
-    expect(ClientCall.toTimeoutString(new Duration(microseconds: 0)), '0u');
-    expect(ClientCall.toTimeoutString(new Duration(microseconds: 107)), '107u');
-    expect(ClientCall.toTimeoutString(new Duration(hours: 2, microseconds: 17)),
-        '7200S');
-    expect(ClientCall.toTimeoutString(new Duration(milliseconds: 1420665)),
-        '1420S');
-    expect(
-        ClientCall.toTimeoutString(new Duration(seconds: 2, microseconds: 3)),
-        '2000m');
-    expect(
-        ClientCall.toTimeoutString(new Duration(seconds: 2, milliseconds: 3)),
-        '2003m');
+  const dummyValue = 0;
+
+  group('Unit:', () {
+    test('Timeouts are converted correctly to header string', () {
+      expect(toTimeoutString(new Duration(microseconds: -1)), '1n');
+      expect(toTimeoutString(new Duration(microseconds: 0)), '0u');
+      expect(toTimeoutString(new Duration(microseconds: 107)), '107u');
+      expect(
+          toTimeoutString(new Duration(hours: 2, microseconds: 17)), '7200S');
+      expect(toTimeoutString(new Duration(milliseconds: 1420665)), '1420S');
+      expect(
+          toTimeoutString(new Duration(seconds: 2, microseconds: 3)), '2000m');
+      expect(
+          toTimeoutString(new Duration(seconds: 2, milliseconds: 3)), '2003m');
+    });
+
+    test('Timeouts are converted correctly from header string', () {
+      expect(fromTimeoutString('1n'), new Duration(microseconds: 1000));
```

Comment:
Add tests for `timeout == null`, timeouts in minutes and hours, and garbage data such as `1`, `1s` and `ab`, and interpreted values such as `-1S`.